libfwupdplugin/fu-hid-device.c

```diff
@@ -146,8 +147,19 @@ fu_hid_device_close (FuUsbDevice *device, GError **error)
 	/* release */
 	if ((priv->flags & FU_HID_DEVICE_FLAG_NO_KERNEL_REBIND) == 0)
 		flags |= G_USB_DEVICE_CLAIM_INTERFACE_BIND_KERNEL_DRIVER;
-	if (!g_usb_device_release_interface (usb_device, priv->interface, flags, error)) {
-		g_prefix_error (error, "failed to release HID interface: ");
+	if (!g_usb_device_release_interface (usb_device, priv->interface, flags, &error_local)) {
+		if (g_error_matches (error_local,
+				     G_USB_DEVICE_ERROR,
+				     G_USB_DEVICE_ERROR_NO_DEVICE) ||
+		    g_error_matches (error_local,
+				     G_USB_DEVICE_ERROR,
+				     G_USB_DEVICE_ERROR_INTERNAL)) {
+			g_debug ("ignoring: %s", error_local->message);
+			return TRUE;
```

Comment:
couldn't this possibly be a situation that the device was USB and unplugged though too?  Shouldn't it only be in certain contexts (such as you know it's about to reboot so it might go away).

Reply:
This error happens when you have a HID device we use with a `FuDeviceLocker` and then get the HID device to reboot. The device returns from the SetReport, reboots -- then we try to call `libusb_release_interface()` on the interface that no longer exists, then `libusb_close()` -- the release of the interface failing as the interface doesn't actually exist any more. I did hope to use just `LIBUSB_ERROR_NO_DEVICE` but libusb returns `LIBUSB_ERROR_NOT_FOUND` which we unhelpfully map to `G_USB_DEVICE_ERROR_INTERNAL` rather than `G_USB_DEVICE_ERROR_NO_DEVICE` or something more helpful.

I'm not sure we can handle this in the plugin code.